plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -171,14 +185,123 @@ public function delete_database_tables() {
 	}
 
 	/**
-	 * Is the data sync between old and new tables currently enabled?
+	 * Is the real-time data sync between old and new tables currently enabled?
 	 *
 	 * @return bool
 	 */
 	public function data_sync_is_enabled(): bool {
 		return 'yes' === get_option( self::ORDERS_DATA_SYNC_ENABLED_OPTION );
 	}
 
+	/**
+	 * Is the background data sync between old and new tables currently enabled?
+	 *
+	 * @return bool
+	 */
+	public function background_sync_is_enabled(): bool {
+		$enabled = $this->data_sync_is_enabled();
+
+		/**
+		 * Toggle for background sync.
+		 *
+		 * This allows for background sync to be enabled even when real-time sync is disabled.
+		 *
+		 * @param bool $enabled True to enabled background sync.
+		 */
+		return apply_filters( 'woocommerce_custom_orders_table_background_sync_enabled', $enabled );
+	}
+
+	/**
+	 * Schedule an event to run background sync when the mode is not set to continuous.
+	 *
+	 * @return void
+	 */
+	private function schedule_background_sync() {
+		$has_scheduled_action = WC()->queue()->search( array(
+			'hook'   => self::BACKGROUND_SYNC_EVENT_HOOK,
+			'status' => array( 'pending' ),
+		) );
+
+		if ( ! $has_scheduled_action ) {
+			/**
+			 * Modify the time interval between background syncs.
+			 *
+			 * Note that the background sync mode must first be set to 'interval' using the
+			 * 'woocommerce_custom_orders_table_background_sync_mode' filter hook before this will work.
+			 *
+			 * @param int $interval The time interval, in seconds, between background syncs. Defaults to 3600 (1 hour).
+			 */
+			$interval = apply_filters( 'woocommerce_custom_orders_table_background_sync_interval', HOUR_IN_SECONDS );
+
+			as_schedule_single_action(
+				time() + $interval,
+				self::BACKGROUND_SYNC_EVENT_HOOK,
+				array(),
+				'',
+				true
+			);
+		}
+	}
+
+	/**
+	 * Remove any pending background sync events.
+	 *
+	 * @return void
+	 */
+	private function unschedule_background_sync() {
+		WC()->queue()->cancel_all( self::BACKGROUND_SYNC_EVENT_HOOK );
+	}
+
+	/**
+	 * Keep background sync running, if it's enabled.
+	 *
+	 * @return void
+	 */
+	private function handle_background_sync() {
+		if ( ! $this->background_sync_is_enabled() ) {
+			$this->unschedule_background_sync();
+			return;
+		}
+
+		/**
+		 * Modify the background sync mode.
+		 *
+		 * @param string $mode The mode for background sync. 'interval' or 'continuous'. Defaults to 'interval'.
+		 */
+		$mode = apply_filters( 'woocommerce_custom_orders_table_background_sync_mode', 'interval' );
+
+		switch ( $mode ) {
+			case 'interval':
```

Comment:
both `interval` and `continous` should be defined as constants.

Reply:
Updated in 898679d4e1b8d7a97136ca0ecf1c446266f62d06